### PR TITLE
`1972` treasury layout update

### LINKED ISSUE #1972: Treasury page - Desktop - Remove assets horizontal scroll
Please update the Assets table in the desktop view to remove the horizontal scroll. 
Browser: Arc (Chromium)

See gif:
Image

### PR DESCRIPTION
**Pull request by Da-Colon**
Fixes #1972 

The rework was a little less invasive as I thought. Was able to achieve the correct layout using `Grid` component. as well as preserve the mobile screen changes easier.

## Notable changes:
- Switch to grid for main layout (using minmax)
- Separate NFT and Coin components to separate component to easier read file.

## Extra note
- I did not change any logic or reviewed deeper into the components themselves just focused on UI updates as needed. 

> Screenshot is 1280x1260
localhost_3000_treasury_dao=eth_0x36bD3044ab68f600f6d3e081056F34f2a58432c4

### CODE REVIEW

**Comment by Da-Colon** [comment]
@decentdao/design @decentdao/product.

One noticeable difference between the figma designs and this implementation.
> Figma
image

> This PR
image

This id is quite long `1125995687427581060846379258837315132058982026049890980521328335525525464358`

So I used a truncate function to shorten it. with that I also added space between the name and id, instead of pushing it together (which makes things really tricky as far as spacing between longer NFT names. Thoughts on this change?

**Comment by tomstuart123** [comment]
@Da-Colon - the removal of side scroll works well for large mid and small screens

however, this new PR seems to be creating some issues with the Txs piece.

See below for what I see in the new PR
Screenshot 2024-06-20 at 14 59 47

v.s. what I see in app.dev

Screenshot 2024-06-20 at 14 59 51

Could this be because of the use of grid makes sure the tx hash is kept in (three columns). Before we cut the Tx hash on small devices (i.e. 2 columns)

**Comment by Da-Colon** [comment]
> @Da-Colon - the removal of side scroll works well for large mid and small screens
> 
> however, this new PR seems to be creating some issues with the Txs piece.
> 
> See below for what I see in the new PR
> Screenshot 2024-06-20 at 14 59 47
> 
> v.s. what I see in app.dev
> 
> Screenshot 2024-06-20 at 14 59 51
> 
> Could this be because of the use of grid makes sure the tx hash is kept in (three columns). Before we cut the Tx hash on small devices (i.e. 2 columns)
> 

O weird. My changes shouldn't have affected that. (Nor did I notice that last night.) I will take a look and get fixed up

**Comment by Da-Colon** [comment]
@tomstuart123 Oookay, I got it. I saw my mistakes here and reenabled scrolling for these tables on the smallest screens. Should be good to go to relook at.

**Comment by tomstuart123** [comment]
good call @Da-Colon . Transactions hashes work better now and I like adding back in the side scroll back in for the other sections too as they were feeling a bit tight 

Approved

**Review by tomstuart123 — APPROVED** [review]

File: src/pages/daos/[daoAddress]/treasury/index.tsx [hunk 1]
```diff
@@ -52,84 +50,64 @@ export default function Treasury() {
...
-        gap="1rem"
-        flexWrap="wrap"
+      <Grid
+        templateAreas={{
+          base: `"assets"
+          "transactions"`,
+          lg: `"transactions assets"`,
+        }}
+        gap={{ base: '1rem', lg: '2rem' }}
+        gridTemplateColumns={{ base: `1fr`, lg: `minmax(718px, 736px) fit-content` }}
```
[the remark this answers is not in the document]
Comment (adamgall) [reply src/pages/daos/[daoAddress]/treasury/index.tsx:60]: +1

[the remark this answers is not in the document]
Comment (Da-Colon) [reply src/pages/daos/[daoAddress]/treasury/index.tsx:60]: Those values come from https://github.com/orgs/decentdao/projects/32/views/3?pane=issue&itemId=63739793

**Comment by Da-Colon** [comment]
> good call @Da-Colon . Transactions hashes work better now and I like adding back in the side scroll back in for the other sections too as they were feeling a bit tight
> 
> Approved

Blah Sorry yall. made a few more adjustments will push up shortly. Found some stuff I wasn't happy with and almost done fixing it. wasn't quite there yet.

**Pull request merged**